substrate/frame/balances/src/lib.rs

```diff
@@ -1082,7 +1087,12 @@ pub mod pallet {
 					frame_system::Pallet::<T>::dec_consumers(who);
 				}
 				if !did_consume && does_consume {
-					frame_system::Pallet::<T>::inc_consumers(who)?;
+					if force_consumer_bump {
```

Comment:
personally, I would always go with `inc_consumers_without_limit`. adding an extra option introduces some complexity without fully eliminating the chance of the consumer ref count exceeding its maximum (+1 only). also, we already use `inc_consumers_without_limit` in other parts of the codebase.

Reply:
Hmm good point, I wasn't sure about the usage of this. 

- The argument to keep it as-is to merely backwards compatibility -- I didn't want to change the behavior of `Reserve` and so on
- But then again, system level operations can and should always use `inc_consumers_without_limit`.

I am happy to go either way (making everything `inc_consumers_without_limit` will simplify the PR indeed), and will spend a bit more time seeing what other pallets are doing before merging.